server.go

```diff
@@ -409,6 +409,16 @@ func (s *ServerImpl) SetMainByPort(port int) error {
 	return s.reloadHaproxy(port)
 }
 
+func (s *ServerImpl) SetMainById(id string) error {
+	for port, deployId := range s.config.Ports {
+		if deployId == id {
+			return s.reloadHaproxy(port)
```

Comment:
The RPC request could have a boolean "force" value (set by the client based on flag) - if it's not true, then the request fails unless there's a 200 OK status server running on that port.